### PR TITLE
Added gradient descent

### PR DESCRIPTION
**Pull request by anesh612**
Here is my contribution.

### CODE REVIEW

File: algorithms/machine_learning/Gradient-Descent/gradientdescent.m [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+function Theta = gradientdescent(x, Y, Theta, Alpha, noi)
+
+    n = length(Y); % Number of training examples. 
+
+    for i = 1:noi 
+
+        theta_1 = Theta(1) - Alpha * (1 / n) * sum(((x * Theta) - Y) .* x(:, 1)); % Temporary variable to simultaneously update theta_0 but i have used 1 to avoid confusion since indexing in MATLAB/Octave starts from 1.
+    
+        theta_2 = Theta(2) - Alpha * (1 / n) * sum(((x * Theta) - Y) .* x(:, 2)); % Temporary variable to simultaneously update theta_1.
```
Comment (cozek) [inline algorithms/machine_learning/Gradient-Descent/gradientdescent.m:19]: Please break up the long comment into multiple lines.

Comment (anesh612) [reply algorithms/machine_learning/Gradient-Descent/gradientdescent.m]: Okay i have made both changes in second commit.

**Comment by cozek** [comment]
Nice work!

**Comment by anesh612** [comment]
My commits are not being counted in my contributions as of now.

**Pull request merged**